### PR TITLE
Mutation optimized by caching the index of clipping view

### PR DESCRIPTION
**Pull request by cyanglaz**
Caching index of the clipping view to make the search index 0(1) instead of 0(n)

https://github.com/flutter/flutter/issues/35023

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h [hunk 1]
```diff
@@ -113,6 +113,12 @@ class FlutterPlatformViewsController {
   std::map<std::string, fml::scoped_nsobject<NSObject<FlutterPlatformViewFactory>>> factories_;
   std::map<int64_t, fml::scoped_nsobject<NSObject<FlutterPlatformView>>> views_;
   std::map<int64_t, fml::scoped_nsobject<FlutterTouchInterceptingView>> touch_interceptors_;
+  // Mapping a platform view ID to the index of the root_view under `flutter_view_.subviews`
+  //
+  // 0 index means represents the first item in the `flutter_view_.subviews`.
+  // See also:
+  // * `root_views_`
+  std::map<int64_t, int64_t> root_views_indices_;
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h]: nit: maybe view_positions_ (skimming on a high level here I think it's easier to think of this as the position of the platform view amongst the children of flutter_view more details about the view chains are in the comment)

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h [hunk 2]
```diff
@@ -113,6 +113,12 @@ class FlutterPlatformViewsController {
   std::map<std::string, fml::scoped_nsobject<NSObject<FlutterPlatformViewFactory>>> factories_;
   std::map<int64_t, fml::scoped_nsobject<NSObject<FlutterPlatformView>>> views_;
   std::map<int64_t, fml::scoped_nsobject<FlutterTouchInterceptingView>> touch_interceptors_;
+  // Mapping a platform view ID to the index of the root_view under `flutter_view_.subviews`
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h]: alternative suggestion:
Maps a platform view's root view to its position amongst `flutter_view_.subviews`.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 3]
```diff
@@ -306,13 +309,14 @@
...
+
   if (currentClippingCount != previousClippingCount) {
     clip_count_[view_id] = currentClippingCount;
     // If we have a different clipping count in this frame, we need to reconstruct the
     // ClippingChildView chain to prepare for `ApplyMutators`.
     UIView* oldPlatformViewRoot = root_views_[view_id].get();
-    UIView* newPlatformViewRoot =
-        ReconstructClipViewsChain(currentClippingCount, touchInterceptor, oldPlatformViewRoot);
+    UIView* newPlatformViewRoot = ReconstructClipViewsChain(currentClippingCount, touchInterceptor,
+                                                            oldPlatformViewRoot, view_id);
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: nit: feels like view_id is enough information to get the touchInterceptor and the oldPlatformViewRoot, may be cleaner to either just pass view_id, or alternatively to pass the view's position.

**Comment by cbracken** [comment]
@cyanglaz @amirh any plans to continue with this PR?

**Comment by cyanglaz** [comment]
@cbracken Yes. The PR is currently blocked by integration test on platform views which we are planning to do after the common platform view factoring.

**Comment by cyanglaz** [comment]
Closing this PR for now until we have iOS platform view E2E test.

**Pull request closed without merging**